The patch series taken from upstream unconditionally changes the stat definitions
in src/unix/linux_like/linux/gnu/b32/x86/mod.rs, since we don't enable time64
on i386 this causes us problems. This patch puts back the old definitions and
adds conditionals.

Index: rust-libc-0.2.153/src/unix/linux_like/linux/gnu/b32/x86/mod.rs
===================================================================
--- rust-libc-0.2.153.orig/src/unix/linux_like/linux/gnu/b32/x86/mod.rs
+++ rust-libc-0.2.153/src/unix/linux_like/linux/gnu/b32/x86/mod.rs
@@ -2,6 +2,7 @@ pub type c_char = i8;
 pub type wchar_t = i32;
 pub type greg_t = i32;
 
+#[cfg(gnu_time64_abi)]
 pub type statfs64 = statfs;
 
 s! {
@@ -130,7 +131,50 @@ s! {
         __unused1: ::c_ulong,
         __unused2: ::c_ulong
     }
+}
 
+#[cfg(not(gnu_time64_abi))]
+s! {
+    pub struct stat64 {
+        pub st_dev: ::dev_t,
+        __pad1: ::c_uint,
+        __st_ino: ::ino_t,
+        pub st_mode: ::mode_t,
+        pub st_nlink: ::nlink_t,
+        pub st_uid: ::uid_t,
+        pub st_gid: ::gid_t,
+        pub st_rdev: ::dev_t,
+        __pad2: ::c_uint,
+        pub st_size: ::off64_t,
+        pub st_blksize: ::blksize_t,
+        pub st_blocks: ::blkcnt64_t,
+        pub st_atime: ::time_t,
+        pub st_atime_nsec: ::c_long,
+        pub st_mtime: ::time_t,
+        pub st_mtime_nsec: ::c_long,
+        pub st_ctime: ::time_t,
+        pub st_ctime_nsec: ::c_long,
+        pub st_ino: ::ino64_t,
+    }
+
+    pub struct statfs64 {
+        pub f_type: ::__fsword_t,
+        pub f_bsize: ::__fsword_t,
+        pub f_blocks: u64,
+        pub f_bfree: u64,
+        pub f_bavail: u64,
+        pub f_files: u64,
+        pub f_ffree: u64,
+        pub f_fsid: ::fsid_t,
+        pub f_namelen: ::__fsword_t,
+        pub f_frsize: ::__fsword_t,
+        pub f_flags: ::__fsword_t,
+        pub f_spare: [::__fsword_t; 4],
+    }
+}
+
+#[cfg(gnu_time64_abi)]
+s! {
     pub struct stat64 {
         pub st_dev: ::dev_t,
         pub st_ino: ::ino_t,
@@ -152,7 +196,9 @@ s! {
         pub st_ctime_nsec: ::c_long,
         __pad3: i32,
     }
+}
 
+s! {
     pub struct statvfs64 {
         pub f_bsize: ::c_ulong,
         pub f_frsize: ::c_ulong,
